ibis/common/parsing.py

```diff
@@ -0,0 +1,40 @@
+import ast
+import re
+
+import parsy
+
+_STRING_REGEX = """('[^\n'\\\\]*(?:\\\\.[^\n'\\\\]*)*'|"[^\n"\\\\"]*(?:\\\\.[^\n"\\\\]*)*")"""  # noqa: E501
+
+SPACES = parsy.regex(r'\s*', re.MULTILINE)
+
+
+def spaceless(parser):
+    return SPACES.then(parser).skip(SPACES)
+
+
+def spaceless_string(*strings: str):
+    return spaceless(
+        parsy.alt(
+            *(parsy.string(string, transform=str.lower) for string in strings)
+        )
+    )
+
+
+RAW_NUMBER = parsy.digit.at_least(1).concat()
```

Comment:
This should really be `parsy.decimal_digit`, since `digit` includes all sorts of things like `²` that Python cannot turn into an `int`:

```
In [6]: import parsy

In [7]: parsy.digit.parse('²')
Out[7]: '²'

In [8]: parsy.decimal_digit.parse('²')
ParseError: expected '[0123456789]' at 0:0

In [9]: parsy.decimal_digit.parse('2')
Out[9]: '2'
```

Reply:
Updated.